sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -14,18 +14,54 @@ import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
 
 function loadEnvironmentProxyValue(): string | undefined {
+  let httpProxy = null;
+  let httpsProxy = null;
+  let allProxy = null;
+  let noProxy = null;
+
   if (!process) {
     return undefined;
   }
 
   if (process.env[Constants.HTTPS_PROXY]) {
```

Comment:
Seems like we have a lot of code that calls toLowerCase().  Should we convert them all to lower at the top or in a separate method so the following code is clean?